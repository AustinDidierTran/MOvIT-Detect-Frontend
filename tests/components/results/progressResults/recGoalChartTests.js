/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import React from 'react';
import PropTypes from 'prop-types';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import RecGoalChart from '../../../../src/components/results/progressResults/recGoalChart';

Enzyme.configure({ adapter: new Adapter() });

describe('RecGoalChart Tests', () => {
  const data = {
    labels: ['label'],
    datasets: [
      {
        label: 'test',
        data: [1],
      },
    ],
  };
  const props = {
    condition: true,
    title: 'test',
    goalTitle: 'test',
    recTitle: 'test',
    goalData: data,
    recData: data,
  };

  it('should match snapshot', () => {
    const wrapper = shallow(<RecGoalChart {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have proptypes', () => {
    const actualValue = RecGoalChart.propTypes;

    const expectedValue = {
      condition: PropTypes.bool.isRequired,
      title: PropTypes.string.isRequired,
      goalTitle: PropTypes.string.isRequired,
      recTitle: PropTypes.string.isRequired,
      goalData: PropTypes.object,
      recData: PropTypes.object,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });
});
