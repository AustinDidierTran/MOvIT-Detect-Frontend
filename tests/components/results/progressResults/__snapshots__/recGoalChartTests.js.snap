// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecGoalChart Tests should match snapshot 1`] = `
<div>
  <div>
    <CustomCard
      element={
        <div>
          <h4>
            test
          </h4>
          <Chart
            className={null}
            data={
              Object {
                "datasets": Array [
                  Object {
                    "data": Array [
                      1,
                    ],
                    "label": "test",
                  },
                ],
                "labels": Array [
                  "label",
                ],
              }
            }
            height={null}
            id={null}
            options={
              Object {
                "legend": Object {
                  "onHover": [Function],
                },
              }
            }
            style={null}
            type="line"
            width={null}
          />
          <hr />
          <h4>
            test
          </h4>
          <Chart
            className={null}
            data={
              Object {
                "datasets": Array [
                  Object {
                    "data": Array [
                      1,
                    ],
                    "label": "test",
                  },
                ],
                "labels": Array [
                  "label",
                ],
              }
            }
            height={null}
            id={null}
            options={
              Object {
                "legend": Object {
                  "onHover": [Function],
                },
              }
            }
            style={null}
            type="line"
            width={null}
          />
        </div>
      }
      header={
        <div>
          <h2
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            test
          </h2>
        </div>
      }
    />
  </div>
</div>
`;
